protostuff-api/src/main/java/io/protostuff/StringSerializer.java

```diff
@@ -318,7 +318,7 @@ public static LinkedBuffer writeDouble(final double value, final WriteSession se
     /**
      * Computes the size of the utf8 string beginning at the specified {@code index} with the specified {@code length}.
      */
-    public static int computeUTF8Size(final String str, final int index, final int len)
+    public static int computeUTF8Size(final CharSequence str, final int index, final int len)

```

Comment:
One noticeable difference is there is an initial slowness in the serialization 35.5-41 ms
then it seems to go back to normal ranges, (after JIT hits in ) if we can assume we are running only on JIT capable JVMs